### PR TITLE
Protect against SQL injections

### LINKED ISSUE #560: The Sequelize query builder could accept SQL injections
Since it's using raw queries, but they aren't escaped properly: https://github.com/fix/ark-core/blob/575ae88505a1a92a648825f9069c2ff381e79280/packages/core-database-sequelize/lib/query-builder.js

It could be fixed using replacements, probably: http://docs.sequelizejs.com/manual/tutorial/raw-queries.html

### PR DESCRIPTION
**Pull request by supaiku0**
## Proposed changes

Proposed solution to fix #560.

The PR basically consists of the following:
1. Turn all column names into snake_case, via migration scripts (Is this the right way?)
and add field mappings to the models
2. Adjust raw sql to use new column names
3. Finally make use Sequelize replacements
4. Fix unit tests / regressions

The column renaming is done to follow the PostgreSQL standard naming conventions and to ultimately get rid of quoted column names in raw queries which don't work nicely together with Sequelize's escaping mechanism.

## Types of changes

What types of changes does your code introduce?

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Refactoring (improve a current implementation without adding a new feature or fixing a bug)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build (changes that affect the build system)
- [ ] Docs (documentation only changes)
- [ ] Test (adding missing tests or fixing existing tests)
- [ ] Other... Please describe:

## Further comments
There's most likely a lot of random breakage, since I didn't test it yet beyond `[ERROR]   : Block download: Peer 167.114.29.51 is unresponsive`, but up to that point it's working. :stuck_out_tongue: 

-  I'm not sure if it makes sense to use migration scripts since it's a big change, why not just start over completely fresh?
- Unit tests need to be fixed
- Code inconsistently uses blockId / blockid, don't know the implications yet, should it be refactored to blockId?
- I guess the SQLite webhooks db needs to be adapted as well?
- Instead of [replacements, bind parameters](http://docs.sequelizejs.com/manual/tutorial/raw-queries.html) could also be used
- Please leave your feedback

### CODE REVIEW

**Comment by faustbrian** [comment]
@supaiku0 There seems to be a conflict. I will try to find time to test it this weekend. @alexbarnsley could you also take a look if you find a minute?

**Comment by supaiku0** [comment]
@faustbrian Thanks, I rebased onto the master branch. Should work again.

**Comment by supaiku0** [comment]
I'm working on the unit tests of `core-database-sequelize` which exposed some quirks. Please wait until I updated the branch before you give it a try.

**Comment by supaiku0** [comment]
The majority of the unit tests pass again (only 2 fails instead of 3 as on master). I hope that means there aren't any major regressions left.

Just hit me on Slack or gitter if realtime collaboration is desired.

File: packages/core-database-sequelize/lib/query-builder/sql-builder.js [hunk 1]
```diff
@@ -67,13 +67,6 @@ class SqlBuilder {
       }
 
       if (['IN', 'NOT IN'].includes(item.operator)) {
-        if (Array.isArray(item.value)) {
```
Comment (faustbrian) [inline packages/core-database-sequelize/lib/query-builder/sql-builder.js]: Does Sequelize automatically join the `item.value` array with `,`?

Comment (faustbrian) [reply packages/core-database-sequelize/lib/query-builder/sql-builder.js]: Alright, will have time to test it tomorrow.

**Comment by supaiku0** [comment]
@fix @kristjank I'll rebase the changes onto master later today, can someone of you do a review then?

**Comment by supaiku0** [comment]
@fix @kristjank Ping :rocket:

**Comment by kristjank** [comment]
@supaiku0 very nice PR 🎢 🎉  
- did you sync from 0, was all 🚦 green :) 
- will spin up a node on this PR also... 
more feedback incoming...

File: packages/core-database-sequelize/lib/query-builder/sql-builder.js [hunk 2]
```diff
@@ -33,7 +36,7 @@ class SqlBuilder {
    */
   __buildSelect (clauses) {
     const columns = clauses.select.columns
-      .map(column => escape(column))
+      .map(column => column)
```
Comment (j-a-m-l) [inline packages/core-database-sequelize/lib/query-builder/sql-builder.js]: This line doesn't do anything

**Comment by supaiku0** [comment]
@j-a-m-l No, I don't mind. Will do it tomorrow.

File: packages/core-database-sequelize/lib/query-builder/sql-builder.js (hunk 2 above)
Comment (supaiku0) [reply packages/core-database-sequelize/lib/query-builder/sql-builder.js]: The line after this one neither. :yum:

**Comment by supaiku0** [comment]
@j-a-m-l I added some basic tests. Feel free to provide more examples. 

I wonder why the drop table one doesn't "work" on master, the generated SQL statement definitely drops the table when I execute it manually on a test SQLite db. Any idea? :thinking:

**Comment by alexbarnsley** [comment]
@supaiku0, does it fail when running on postgres? We noticed that sqlite and postgres sometimes behaved differently

**Comment by supaiku0** [comment]
Rebased.

No point in putting any more effort into the unit tests since they don't even run against Postgres. But I talked with Brian, if I find the time I will look into dropping SQLite for the tests. Afterwards we should revise the unit tests.

**Pull request merged**